R/choices_selected.R

```diff
@@ -130,10 +130,18 @@ choices_selected <- function(choices,
                              selected = if (inherits(choices, "delayed_data")) NULL else choices[1],
                              keep_order = FALSE,
                              fixed = FALSE) {
-  stopifnot(is.atomic(choices) || inherits(choices, "delayed_data"))
-  stopifnot(is.atomic(selected) || inherits(selected, "delayed_data") || inherits(selected, "all_choices"))
-  checkmate::assert_flag(keep_order)
-  checkmate::assert_flag(fixed)
+  checkmate::assert(
+    checkmate::check_atomic(choices),
+    checkmate::check_class(choices, "delayed_data")
+  )
+  checkmate::assert(
+    checkmate::check_atomic(selected),
+    checkmate::check_multi_class(selected, c("delayed_data", "all_choices"))
+  )
+  checkmate::assert(
+    checkmate::check_flag(keep_order),
+    checkmate::check_flag(fixed)
+  )
```

Comment:
Why the change in this? it wasn't an `OR` operation

```suggestion
  checkmate::assert_flag(keep_order)
  checkmate::assert_flag(fixed)
```

Reply:
True, I must have mixed it up while implementing this for `||` cases. Thanks for catching it!